http/src/main/scala/org/http4s/blaze/http/http2/Http2StageTools.scala

```diff
@@ -0,0 +1,56 @@
+package org.http4s.blaze.http.http2
+
+import org.http4s.blaze.http.util.HeaderNames
+
+import scala.annotation.tailrec
+import scala.collection.BitSet
+
+
+object Http2StageTools {
+  // Request pseudo headers
+  val Method = ":method"
+  val Scheme = ":scheme"
+  val Path   = ":path"
+  val Authority = ":authority"
+
+  // Response pseudo header
+  val Status = ":status"
+  val TE = "te"
+  val Connection = HeaderNames.Connection
+  val ContentLength = HeaderNames.ContentLength
+
+  /** Check of the header name is a valid http2 header name.
+    * Pseudo headers are considered invalid and should be screened before hand.
+    */
+  def validHeaderName(str: String): Boolean = {
```

Comment:
Is this faster than `str.nonEmpty && str.forall(...)`?  (Probably, but you reinvented a wheel.)

Reply:
Won't the `String.forall` result in boxing each `Char`? That is what I was trying to avoid.